### PR TITLE
Api - Authorization

### PR DESCRIPTION
**Pull request by HansDahle**

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Logic/Requests/Commands/ContractorPersonnelRequest/CheckUserAccess.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    if (dbRequest == null)
+                        return false;
+
+
+                    var contract = await orgResolver.ResolveContractAsync(dbRequest.OrgProjectId, dbRequest.OrgContractId);
+
+                    if (contract == null)
+                        return false;
+
+                    if (request.Editor.AzureUniqueId == null)
```
Comment (leifbjarte) [inline src/backend/api/Fusion.Resources.Logic/Requests/Commands/ContractorPersonnelRequest/CheckUserAccess.cs]: Move this check to the top?

File: src/backend/api/Fusion.Resources.Api/Authorization/Requirements/ContractRole.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿using Fusion.AspNetCore.FluentAuthorization;
+
+namespace Fusion.Resources.Api.Authorization
+{
+    public class ContractRole : IReportableAuthorizationRequirement
```
[the remark this answers is not in the document]
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Authorization/Requirements/ContractRole.cs]: Mmm, was thinking the same, but should do that in the library where the interface is i think. 
Will refactor.

File: src/backend/api/Fusion.Resources.Api/Authorization/Handlers/RequestAccessAuthHandler.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                case RequestAccess.AccessType.Workflow:
+                    await EvaluateWorkflowAccessAsync(context, requirement, resource);
+                    break;
+            }
+        }
+
+        private async Task EvaluateWorkflowAccessAsync(AuthorizationHandlerContext context, RequestAccess requirement, QueryPersonnelRequest resource)
+        {
+            var hasNaturalAccess = await mediator.Send(new ContractorPersonnelRequest.CheckUserAccess(resource.Id));
```
[the remark this answers is not in the document]
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Authorization/Handlers/RequestAccessAuthHandler.cs]: Heh can agree on that.. had some back and forth moments on  how to do this..

File: src/backend/api/Fusion.Resources.Api/Authorization/Requirements/ContractRole.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Authorization/Requirements/ContractRole.cs]: Found this type was created in the Fusion.Integration.Authorization library.

**Pull request merged**